src/test/java/org/eclipse/pass/schema/SchemaFetcherTest.java

```diff
@@ -0,0 +1,168 @@
+import org.dataconservancy.pass.model.Repository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class SchemaFetcherTest {
+
+    private PassClient passClientMock;
+    private Repository repository1mock;
+    private Repository repository2mock;
+
+    @BeforeEach
+    void setup() {
+        passClientMock = Mockito.mock(PassClient.class);
+        repository1mock = Mockito.mock(Repository.class);
+        repository2mock = Mockito.mock(Repository.class);
+    }
+
+    @Test
+    void getSchemaFromUriTest() throws StreamReadException, DatabindException, IOException, URISyntaxException {
+        SchemaFetcher s = new SchemaFetcher(passClientMock);
+        URI uri = new URI("https://example.com/pass-metadata-schemas/example/schemas/schema1.json");
+        String expectedJsonString = "{\r\n" + "    \"$schema\": \"http://example.org/example/schemas/schema\",\r\n"
+                + "    \"$id\": \"http://example.org/example/schemas/foo\",\r\n" + "    \"title\": \"foo\",\r\n"
+                + "    \"description\": \"foo schema\",\r\n" + "    \"$comment\": \"one\",\r\n"
+                + "    \"a\": \"1\",\r\n" + "    \"x\": {\r\n" + "        \"title\": \"X\",\r\n"
+                + "        \"description\": \"a letter\",\r\n" + "        \"$comment\": \"displays good\",\r\n"
+                + "        \"type\": \"letter\"\r\n" + "    },\r\n" + "    \"array\": [\"a\", \"b\", \"c\"],\r\n"
+                + "    \"aa\": \"b\",\r\n" + "    \"cc\": [\"d\", \"e\"]\r\n" + "}";
+        ObjectMapper map = new ObjectMapper();
+        JsonNode expected = map.readTree(expectedJsonString);
+        assertEquals(expected, s.getSchemaFromUri(uri));
+    }
+
+    @Test
+    void getRepositorySchemasTest() throws StreamReadException, DatabindException, IOException, URISyntaxException {
+        when(passClientMock.findByAttribute(Repository.class, "@id", new URI("repository1")))
+                .thenReturn(new URI("uri_to_repository1"));
+        when(passClientMock.findByAttribute(Repository.class, "@id", new URI("repository2")))
+                .thenReturn(new URI("uri_to_repository2"));
+        when(passClientMock.readResource(new URI("uri_to_repository1"), Repository.class)).thenReturn(repository1mock);
+        when(passClientMock.readResource(new URI("uri_to_repository2"), Repository.class)).thenReturn(repository2mock);
+        List<URI> r1_schemas_list = Arrays.asList(new URI("/example/schemas/schema1.json"),
+                new URI("/example/schemas/schema2.json"));
+        List<URI> r2_schemas_list = Arrays.asList(new URI("/example/schemas/schema2.json"),
+                new URI("/example/schemas/schema3.json"), new URI("/example/schemas/schema_to_deref.json"));
+        when(repository1mock.getSchemas()).thenReturn(r1_schemas_list);
+        when(repository2mock.getSchemas()).thenReturn(r2_schemas_list);
+        SchemaFetcher s = new SchemaFetcher(passClientMock);
```

Comment:
I'd turn the SchemaFetcher into a class field and initialize it in the setup method.